### ISSUE TITLE
"Final test ELBO is much lower than the max test ELBO." in a seemingly correct result.

### ISSUE THREAD

**Issue by ZhangMH2000** [broadinstitute/CellBender#294]
Dear developer,

Thank you for developing such a great tool! I have encountered some issues when reading the report html, so I carefully read the issues #220 and #276 . The default result was quite unsatisfactory, and I set the `--learning-rate` 1e-5, `--expected-cells` 10000 and `--total-droplets-included` 50000. The result was better with no warnings, but still not satisfactory:
![image](https://github.com/broadinstitute/CellBender/assets/117978582/475988d0-5042-4964-8d36-03ff9d0d6edb)

So I reduced `--learning-rate` to 1e-6. This time the plot seems to be perfect, however, there is an unexpected warning:
![image](https://github.com/broadinstitute/CellBender/assets/117978582/fafbfded-f380-47b3-ad03-d5cdec582226)
![image](https://github.com/broadinstitute/CellBender/assets/117978582/fe55b879-5e07-4b6b-873f-1182e85d76d7)

```html
Automated assessment --------

We hope to see the test ELBO follow the training ELBO, increasing almost monotonically (though there will be deviations, and that is expected). There may be a large gap, and that is okay. However, this curve ends with a low test ELBO compared to the max test ELBO value during training. The output could be suboptimal.
Summary:

This is slightly unusual behavior, and a reduced --learning-rate might be indicated. Consider re-running with half the current learning rate to compare the results.

Summary of warnings:
Final test ELBO is much lower than the max test ELBO.
``` 

I myself did not observe that the final test ELBO in the warning is much lower than the max test ELBO. I further set the `--epochs` as 250. This time the report has no more warnings, but as far as I'm concerned, I don't think the QC plot is as good as the previous one:
![image](https://github.com/broadinstitute/CellBender/assets/117978582/29adb9c7-f462-460b-aff3-8e62f1022d20)
![image](https://github.com/broadinstitute/CellBender/assets/117978582/e32a2baa-cb43-49f4-9f00-e9c00b9dc35e)

So I'd like to consult your professional opinion, which one do you think is best suited for this dataset? If the ELBO plot meets the requirements: ELBO values are more than a few hundred, both curves increasing and reaching a stable plateau to convergence, and test ELBO didn't dip way back down at the end, can I ignore this warning and use this output? Thank you!

**Comment by sjfleming** [comment]
Hi @ZhangMH2000 , in general I would be slightly skeptical about the automated assessments and warnings ... we try our best to make them accurate, but there are cases where they won't be totally accurate.

To me, I actually think the `--learning-rate 1e-5` result looks good.  This one is 1e-5, if I understand correctly:
![image](https://github.com/broadinstitute/CellBender/assets/10214815/abb2f918-40a4-4072-bb42-7bd1b5686c92)

I am a little concerned about using a learning rate that is much lower than 1e-5.  I think 1e-5 is the lowest I would go unless I had a really good reason.  The basic rule of thumb should be to try to (nearly) push the learning rate as high as you can without causing a problem.  While "causing a problem" is somewhat subjective, I think the above learning curve looks fine.  I suppose the test ELBO jumps around a bit, but not more than I would expect.  And I think the train ELBO looks great.

> If the ELBO plot meets the requirements: ELBO values are ... increasing and reaching a stable plateau to convergence, and test ELBO didn't dip way back down at the end, can I ignore this warning and use this output?

In general yes, if you think the learning curve looks alright and has no major red flags (like huge downward spikes, a large dip down at the end, or just a complete failure to converge toward a stable value), then you can ignore warnings and use the output.

The only times I've seen the output actually compromised in a way that was concerning was when the ELBO value dips way down at the end.  In those cases, it's almost always a problem with cell calling.  Those are often cases where learning gets thrown off track and CellBender says "everything is a cell!"

**Comment by ZhangMH2000** [comment]
Hi @sjfleming, thank you for your reply! I have checked my results to ensure I use the most appropriate learning rate. In some cases, a learning rate of 1e-5 still causes problems, so I have to use 1e-6. However, when you told me that ‘The basic rule of thumb should be to try to (nearly) push the learning rate as high as you can without causing a problem.’, I found that many samples processed with `--learning-rate` 1e-4 have almost only one large blob in the PCA plot, but samples with `--learning-rate` 1e-5 have clusters that correspond to different cell types. So I was confused about which one I should use. I think better clustering can be achieved after appropriate dimensionality reduction and clustering. I look forward to your reply!

**Comment by JordanCTLin** [comment]
Any update on this PCA question? Thanks
